File: 2021_random_set_self_practice/using_python/1465.py
import sys
sys.stdin = open("input.txt", "r")
input = sys.stdin.readline

SZ = 10000001


def cnt_primelike(p, L, R):
    cnt = 0
    num = p ** 2
    while num < L:
        num *= p
    while num <= R:
        cnt += 1
        R = num * p
    return cnt


if __name__ == "__main__":
    is_prime = [True for _ in range(SZ)]
    for i in range(2, SZ):
        for j in range(i + i, SZ, i):
            if not is_prime[i]: continue

            is_prime[j] = False

    primes = []
    for i in range(2, SZ):
        if is_prime[i]:
            primes.append(i)

    L, R = map(int, input().rstrip().split())
    cnt = 0
    for i in range(len(primes)):
        if primes[i] ** 2 > R:
            break

        cnt += cnt_primelike(primes[i], L, R)
    print(cnt)

File: 2021_random_set_self_practice/using_cpp/1465.cpp
/*
input :
1 1000

output :
25
 */

#include <iostream>
#include <cstring>
#define SZ (10000001)

using namespace std;
using ull = unsigned long long;

// 전역 변수 선언
bool IS_PRIME[SZ];
ull PRIME[SZ];

int cnt_primelike(ull p, ull L, ull R) {
    int cnt = 0;
    ull num = p * p;

    while (num < L) {
        num *= p;
    }
    while (num <= R) {
        cnt++;
        R = R / p;
    }
    return cnt;
}


int main(){
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    cout.tie(nullptr);
    freopen("input.txt", "r", stdin);

    // 소수 전처리
    // 10^7 이하의 소수들
    memset(IS_PRIME, true, SZ);
    for (int i = 2; i < SZ; i++) {
        for (int j = i + i; j < SZ; j += i) {
            if (!IS_PRIME[i]) continue;
            IS_PRIME[j] = false;
        }
    }
    int idx = 0;
    for (int i = 2; i < SZ; i++) {
        if (IS_PRIME[i]) {
            PRIME[idx++] = i;
        }
    }

    ull L, R;
    cin >> L >> R;
    int cnt = 0;
    for (int i = 0; i < idx; i++) {
        if (PRIME[i] * PRIME[i] > R) {
            break;
        }
        cnt += cnt_primelike(PRIME[i], L, R);
    }
    cout << cnt << '\n';

    return 0;
}